demo/midgard-sdk/src/initialization.ts

```diff
@@ -0,0 +1,168 @@
+} from "./constants.js";
+import { StateQueueRedeemer } from "./state-queue.js";
+import { ActiveOperatorMintRedeemer } from "./active-operators.js";
+import { RegisteredOperatorMintRedeemer } from "./registered-operators.js";
+import { RetiredOperatorMintRedeemer } from "./retired-operators.js";
+
+export type InitializationParams = {
+  midgardValidators: MidgardValidators;
+};
+
+export const incompleteInitializationTxProgram = (
+  lucid: LucidEvolution,
+  params: InitializationParams,
+): Effect.Effect<TxBuilder, LucidError | Bech32DeserializationError> =>
+  Effect.gen(function* () {
+    const utxos = yield* Effect.tryPromise({
+      try: () => lucid.wallet().getUtxos(),
+      catch: (e) =>
+        new LucidError({
+          message: "Failed to fetch UTxOs for nonce",
+          cause: e,
+        }),
+    });
+
+    if (utxos.length === 0) {
+      return yield* Effect.fail(
+        new LucidError({
+          message: "No UTxOs available for nonce",
+          cause: "Wallet has no UTxOs",
+        }),
+      );
+    }
+
+    const nonceUtxo = utxos[0];
+    const genesisTime = BigInt(Date.now() + 5 * 60 * 1000);
+    let tx = lucid
+      .newTx()
+      .collectFrom([nonceUtxo])
+      .validTo(Number(genesisTime));
+
+    const hubOracleTx = yield* incompleteHubOracleInitTxProgram(
+      lucid,
+      params.midgardValidators,
+    );
+
+    const stateQueueData: ConfirmedState = {
+      headerHash: GENESIS_HASH_28,
+      prevHeaderHash: GENESIS_HASH_28,
+      utxoRoot: GENESIS_HASH_32,
+      startTime: genesisTime,
+      endTime: genesisTime,
+      protocolVersion: INITIAL_PROTOCOL_VERSION,
+    };
+
+    const stateQueueTx = yield* incompleteInitLinkedListTxProgram(lucid, {
+      validator: params.midgardValidators.stateQueueAuthValidator,
+      data: Data.castTo(stateQueueData, ConfirmedState),
+      redeemer: Data.to("Init", StateQueueRedeemer),
+    });
+
+    const registeredOperatorsTx = yield* incompleteInitLinkedListTxProgram(
+      lucid,
+      {
+        validator: params.midgardValidators.registeredOperatorsAuthValidator,
+        redeemer: Data.to("Init", RegisteredOperatorMintRedeemer),
+      },
+    );
+
+    const activeOperatorsTx = yield* incompleteInitLinkedListTxProgram(lucid, {
+      validator: params.midgardValidators.activeOperatorsAuthValidator,
+      redeemer: Data.to("Init", ActiveOperatorMintRedeemer),
+    });
+
+    const retiredOperatorsTx = yield* incompleteInitLinkedListTxProgram(lucid, {
+      validator: params.midgardValidators.retiredOperatorsAuthValidator,
+      redeemer: Data.to("Init", RetiredOperatorMintRedeemer),
+    });
+
+    const GENESIS_OPERATOR = "0".repeat(56);
+
+    const schedulerGenesisData: SchedulerDatum = {
+      operator: GENESIS_OPERATOR,
+      startTime: genesisTime,
+    };
+
+    const schedulerTx = incompleteSchedulerInitTxProgram(lucid, {
+      validator: params.midgardValidators.schedulerAuthValidator,
+      datum: schedulerGenesisData,
+    });
```

Comment:
This seems to be a withstanding issue: #296 

The spec should get an update, but for the time being, don't attach any datum to signal the initial state:
```suggestion
    const schedulerTx = incompleteSchedulerInitTxProgram(lucid, {
      validator: params.midgardValidators.schedulerAuthValidator,
      datum: undefined, // or whichever null value that makes sense.
    });
```